src/info/repo.rs

```diff
@@ -16,159 +14,150 @@ use time::OffsetDateTime;
-        number_of_commits.to_string()
-    }
+                if is_bot(commit.author, bot_regex_pattern) {
+                    continue;
+                }
+                num_commits += 1;
 
-    pub fn get_authors(
-        &self,
-        number_of_authors_to_display: usize,
-        show_email: bool,
-    ) -> Result<(Vec<Author>, usize)> {
-        let mut author_to_number_of_commits: HashMap<Sig, usize> = HashMap::new();
-        let mut total_nbr_of_commits = 0;
-        let mailmap = self.repo.mailmap()?;
-        for commit in &self.logs {
-            let author = match commit.author_with_mailmap(&mailmap) {
-                Ok(val) => val,
-                Err(_) => commit.author(),
-            };
-            let author_nbr_of_commits = author_to_number_of_commits
-                .entry(Sig::from(author))
-                .or_insert(0);
-            *author_nbr_of_commits += 1;
-            total_nbr_of_commits += 1;
+                let author_nbr_of_commits = author_to_number_of_commits
+                    .entry(Sig::from(mailmap.resolve(commit.author)))
+                    .or_insert(0);
+                *author_nbr_of_commits += 1;
+                total_nbr_of_commits += 1;
+
+                time_of_most_recent_commit.get_or_insert_with(|| commit.time());
+                if commit_iter.peek().is_none() {
+                    time_of_first_commit = commit.time().into();
+                }
+            }
         }
 
         let mut authors_by_number_of_commits: Vec<(Sig, usize)> =
             author_to_number_of_commits.into_iter().collect();
 
-        let number_of_authors = authors_by_number_of_commits.len();
-
+        let total_num_authors = authors_by_number_of_commits.len();
         authors_by_number_of_commits.sort_by(|(_, a_count), (_, b_count)| b_count.cmp(a_count));
 
-        if number_of_authors > number_of_authors_to_display {
-            authors_by_number_of_commits.truncate(number_of_authors_to_display);
-        }
-
         let authors: Vec<Author> = authors_by_number_of_commits
             .into_iter()
             .map(|(author, author_nbr_of_commits)| {
+                let email = author.email;
                 Author::new(
-                    author.name.clone(),
-                    show_email.then(|| author.email),
+                    author.name,
+                    email.into(),
                     author_nbr_of_commits,
                     total_nbr_of_commits,
                 )
             })
+            .take(number_of_authors_to_display)
             .collect();
 
-        Ok((authors, number_of_authors))
+        drop(commit_iter);
+        Ok(Self {
+            repo,
+            git2_repo,
+            authors,
+            total_num_authors,
+            num_commits,
+            time_of_first_commit: time_of_first_commit.expect("at least one commit"),
+            time_of_most_recent_commit: time_of_most_recent_commit.expect("at least one commit"),
```

Comment:
A good catch! `onefetch` runs on so many computers, it might actually hit this case. It's addressed in 6817e48be827d95efa1c3ad87cdc0bfaf3421cd4 . I chose to not make it fatal but instead try to continue.

Reply:
Thank you!